MSAL/src/MSALErrorConverter.m

```diff
@@ -249,8 +250,12 @@ + (NSError *)errorWithDomain:(NSString *)domain
     if (errorDescription) msalUserInfo[MSALErrorDescriptionKey] = errorDescription;
     if (oauthError) msalUserInfo[MSALOAuthErrorKey] = oauthError;
     if (subError) msalUserInfo[MSALOAuthSubErrorKey] = subError;
-    
-    if (underlyingError) msalUserInfo[NSUnderlyingErrorKey] = [MSALErrorConverter msalErrorFromMsidError:underlyingError];
+        
+    if (underlyingError) {
+        msalUserInfo[NSUnderlyingErrorKey] = [MSALErrorConverter msalErrorFromMsidError:underlyingError];
```

Comment:
Sorry Deigo, I didn't get your suggestion. 
I'm adding the error codes in the userInfo and in the internal underlying error (in the blue and in the red area)
<img width="698" alt="Screenshot 2024-09-03 at 17 16 33" src="https://github.com/user-attachments/assets/8b610b46-ded3-4752-96bc-58f35378d91b">

Reply:
Thanks Diego. I noticed that unit tests were not covering all scenarios. I added a new test, so it should be fixed now. 

Line 256-257 is needed in case the error codes are contained in the underlying error but not in the user info. This is case triggered in the `MSIDSilentTokenRequest` class line 375, that is aprt of the refresh token scenario.